File: solutions/typescript/grade-school/1/grade-school.ts
export default class GradeSchool {
  private roster: Map<string, string[]> = new Map();

  studentRoster() {
    const roster = new Map();

    for (const [key, value] of this.roster) {
      roster.set(key, [...value]);
    }

    return roster;
  }

  addStudent(name: string, grade: number) {
    this.roster.set(
      grade.toString(),
      this.studentsInGrade(grade).concat(name).sort()
    );
  }

  studentsInGrade(grade: number): string[] {
    return [...(this.roster.get(grade.toString()) || [])];
  }
}

File: solutions/csharp/grade-school/1/GradeSchool.cs
public class GradeSchool
{

    private readonly Dictionary<string, int> _roster = new();
    
    public bool Add(string student, int grade) => _roster.TryAdd(student, grade);

    public IEnumerable<string> Roster() => _roster
        .OrderBy(x => x.Value)
        .ThenBy(x => x.Key)
        .Select(pair => pair.Key);

    public IEnumerable<string> Grade(int grade) => _roster
        .Where(pair => pair.Value == grade)
        .Select(pair => pair.Key)
        .OrderBy(name => name);
}